<a href="https://colab.research.google.com/github/Anikawi/propuesta_parte1_S100_grupo9/blob/main/Proyecto_Riesgo_Postcompra_Autos_Panama.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Fuente 1: ACODECO
## Datos: reclamaciones del sector automotriz en Panamá
## Periodo inicial de prueba: 2022 - 2023 - 2024 - 2025.

# **Objetivo del análisis**
Analizar los factores asociados al riesgo poscompra dentro del sector automotriz en Panamá, utilizando registros de reclamaciones de agencias de autos, venta de autos usados, talleres de reparación, repuestos y alquiler de autos.

## 1. Importar librerias necesarias



In [1]:
import pandas as pd

# 2. Carga y exploración inicial de los datos



In [2]:
df_2022 = pd.read_csv("decisiondequejas_csv2022.csv", sep=",", encoding="latin1")
df_2023 = pd.read_csv("decision_2023.csv", sep=";", encoding="latin1")
df_2024 = pd.read_csv("decision-de-quejas-2024.csv", sep=";", encoding="latin1")
df_2025 = pd.read_csv("decisionquejas_2025.csv", sep=";", encoding="latin1")

print("2022:", df_2022.shape)
print("2023:", df_2023.shape)
print("2024:", df_2024.shape)
print("2025:", df_2025.shape)

2022: (1056, 17)
2023: (2010, 17)
2024: (1915, 17)
2025: (2063, 17)


## 2.1 Cuáles nombres de columnas no coinciden?

In [3]:
print(df_2022.columns.tolist())
print(df_2023.columns.tolist())
print(df_2024.columns.tolist())
print(df_2025.columns.tolist())

['Departamento  de la Instituci¢n', 'Provincias dela Rep£blica  de Panam\xa0', 'Tipo de Ley', 'Fecha de Entrada de Queja', 'Mes de Entrada de Queja', 'Motivo de Queja', 'Agente Econ¢mico', 'Actividad', 'Marca de Producto', 'Producto', 'Velor de Apertura de Queja', 'Resultado de Queja', 'Efectivo Devuelto al Consumidor (B/.)', 'Fecha de Cierre de Queja', 'Mes Cierre de Queja', 'A¤o Cierre de Queja', 'A¤o Recibido de Queja']
['Departamento  de la Instituci¢n', 'Provincias dela Rep£blica  de Panam\xa0', 'Tipo de Ley', 'Fecha de Entrada de Queja', 'Mes de Entrada de Queja', 'Motivo de Queja', 'Agente Econ¢mico', 'Actividad', 'Marca de Producto', 'Producto', 'Velor de Apertura de Queja', 'Resultado de Queja', 'Efectivo Devuelto al Consumidor (B/.)', 'Fecha de Cierre de Queja', 'Mes Cierre de Queja', 'A¤o Cierre de Queja', 'A¤o Recibido de Queja']
['Departamento  de la Instituci¢n', 'Provincias dela Rep£blica  de Panam\xa0', 'Tipo de Ley', 'Fecha de Entrada de Queja', 'Mes de Entrada de Quej

## 2.2 Renombrar las columnas para que los 3 archivos tengan la misma columna

In [4]:
df_2022.columns = df_2025.columns
df_2023.columns = df_2025.columns
df_2024.columns = df_2025.columns

## 2.3 Combinamos los 3 registros

In [5]:
df = pd.concat(
    [df_2023, df_2023, df_2024, df_2025],
    ignore_index=True
)

df.shape

(7998, 17)

## 2.4 Mostrar los primeros 5 registros

como vemos es un registro que recopila quejas de todo tipo a instituciones de todo tipo. entre ellas incluida AUTOS/AUTOMOVILES que es lo que nos interesa. tiene valores sucios y casillas NaN que hay que descartar o no


In [6]:
df.head()

,Departamento de la Institución,Provincias dela República de Panamá,Tipo de Ley,Fecha de Entrada de Queja,Mes de Entrada de Queja,Motivo de Queja,Agente Económico,Actividad,Marca de Producto,Producto,Velor de Apertura de Queja,Resultado de Queja,Efectivo Devuelto al Consumidor (B/.),Fecha de Cierre de Queja,Mes Cierre de Queja,Año Cierre de Queja,Año Recibido de Queja
0,Decisi¢n,Panam,Ley 45 de 2007,44937,Ene 23,Incumplimiento de contrato,"SOMOS RESERVA PANAMµ, S.A.",Agencias de viajes,No se Registro,No se Registro,1034.05,En proceso,NaN,NaN,NaN,NaN,2023
1,Decisi¢n,Panam,Ley 45 de 2007,44937,Ene 23,Incumplimiento de servicio,"SOMOS RESERVA PANAMµ, S.A.",Agencias de viajes,No se Registro,No se Registro,567.84,Corregidur¡a/Casa de Justicia Comunitaria de Paz,NaN,45208.0,Oct 23,2023.0,2023
2,Decisi¢n,Panam,Ley 45 de 2007,44930,Ene 23,Falta de informaci¢n,INSTITUTO SUPERIOR DE CIENCIAS Y TECNOLOGIA,Universidad,No se Registro,No se Registro,3073.0,Resoluci¢n/Devoluci¢n,3073.0,45146.0,Ago 23,2023.0,2023
3,Decisi¢n,Panam,Ley 45 de 2007,44932,Ene 23,Incumplimiento de servicio,GIRASOL BALBOA/,Decoraci¢n,No se Registro,No se Registro,471.0,Desistimiento/Devoluci¢n,NaN,44981.0,Feb 23,2023.0,2023
4,Decisi¢n,Panam,Ley 45 de 2007,44930,Ene 23,Da¤os y perjuicios,LABORATORIO INTEGRAL DE ANALISIS CLINICO POS I...,Cl¡nicas,No se Registro,No se Registro,61.14,Inadmitida,NaN,44998.0,Mar 23,2023.0,2023


## 2.5 Consultar el formato de cada columna

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7998 entries, 0 to 7997
Data columns (total 17 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   Departamento  de la Institución        7998 non-null   object 
 1   Provincias dela República  de Panamá   7998 non-null   object 
 2   Tipo de Ley                            7998 non-null   object 
 3   Fecha de Entrada de Queja              7998 non-null   object 
 4   Mes de Entrada de Queja                7998 non-null   object 
 5   Motivo de Queja                        7998 non-null   object 
 6   Agente Económico                       7998 non-null   object 
 7   Actividad                              7998 non-null   object 
 8   Marca de Producto                      6691 non-null   object 
 9   Producto                               6991 non-null   object 
 10  Velor de Apertura de Queja             7998 non-null   object 
 11  Resu

# 3. Revisión de variables relevantes para el análisis

## 3.1 Cómo viene la variable obj

In [8]:
df["Resultado de Queja"].value_counts(dropna=False)

,count
Resultado de Queja,
En proceso,4302
Desistimiento,1270
Acuerdo en audiencia/Varios,577
Pendiente constancia desistimiento,473
Desistimiento/Devoluci¢n,290
Acuerdo en audiencia/Devoluci¢n de dinero,183
Corregidur¡a/Casa de Justicia Comunitaria de Paz,115
Desistimiento/Devolución,89
Ministerio P£blico,76


## 3.2 Limpieza de la variable obj



In [9]:
df["Resultado de Queja"] = df["Resultado de Queja"].replace({
    "Desistimiento/Devoluci¢n":                           "Desistimiento/Devolución",
    "Ministerio P£blico":                                 "Ministerio Público",
    "Corregidur¡a/Casa de Justicia Comunitaria de Paz":   "Corregiduría/Casa de Justicia Comunitaria de Paz",
    "Resoluci¢n/Devoluci¢n":                              "Resolución/Devolución",
    "Transacci¢n extrajudicial":                          "Transacción extrajudicial",
    "Resoluci¢n/Reemplazo":                               "Resolución/Reemplazo",
    "Acuerdo en audiencia/Reparaci¢n":                    "Acuerdo en audiencia/Reparación",
    "Resoluci¢n/Reparar":                                 "Resolución/Reparar",
    "Ordena correcci¢n":                                  "Ordena corrección",
    "Acuerdo en audiencia/Aclaraci¢n":                    "Acuerdo en audiencia/Aclaración",
    "Temporal Pendiente de notificaci¢n":                 "Temporal Pendiente de notificación",
    "Acuerdo/Devoluci¢n":                                 "Acuerdo/Devolución",
    "Resoluci¢n no acceder":                              "Resolución no acceder",
    "Acuerdo en audiencia/Devoluci¢n de dinero":          "Acuerdo en audiencia/Devolución de dinero",
    "Acuerdo/Reparaci¢n":                                 "Acuerdo/Reparación",
    "":"",
})

In [10]:
df["Resultado de Queja"].value_counts()

,count
Resultado de Queja,
En proceso,4302
Desistimiento,1270
Acuerdo en audiencia/Varios,577
Pendiente constancia desistimiento,473
Desistimiento/Devolución,379
Acuerdo en audiencia/Devolución de dinero,203
Corregiduría/Casa de Justicia Comunitaria de Paz,121
Ministerio Público,81
Acuerdo en audiencia/Reemplazo,74


## 3.3 Filtrar solo los casos relacionados con automóviles y contar cuántos registros tenemos

In [11]:
df_autos = df[
    (df["Actividad"] == "Agencia de autos") &
    (df["Producto"] == "Automóvil")
].copy()

df_autos.shape

(177, 17)

In [12]:
print("2022:", len(df_2022[
    (df_2023["Actividad"] == "Agencia de autos") &
    (df_2023["Producto"] == "Automóvil") #"Auto"
]))

print("2023:", len(df_2023[
    (df_2023["Actividad"] == "Agencia de autos") &
    (df_2023["Producto"] == "Auto") #"Auto"
]))

print("2024:", len(df_2024[
    (df_2024["Actividad"] == "Agencia de autos") &
    (df_2024["Producto"] == "Automóvil") #"Auto"
]))

print("2025:", len(df_2025[
    (df_2025["Actividad"] == "Agencia de autos") &
    (df_2025["Producto"] == "Automóvil")
]))

2022: 0
2023: 94
2024: 0
2025: 177


/tmp/ipykernel_111560/418014262.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print("2022:", len(df_2022[


In [13]:
for anio, datos in [("2022", df_2022), ("2023", df_2023), ("2024", df_2024), ("2025", df_2025)]:
    print("\n", anio)
    print(datos[
        datos["Producto"].astype(str).str.contains(
            "auto|veh|carro|motor", case=False, na=False
        )
    ]["Producto"].value_counts())


 2022
Producto
Auto               125
Motor de Auto        2
Bater¡a de auto      1
Name: count, dtype: int64

 2023
Producto
Auto                281
Motor de Auto         7
Bater¡a de auto       6
Pantalla de auto      2
Name: count, dtype: int64

 2024
Producto
Auto               336
Bater¡a de auto      6
Name: count, dtype: int64

 2025
Producto
Automóvil          478
Batería de auto      6
Motor de Auto        2
Name: count, dtype: int64


Nota: se realizo el analisis mas profundo porque en 3 años es extraño que no exista registro de quejas. se encontro que :  

*   2022: Producto = Auto
*   2024: Producto = Auto
*   2025: Producto = Automóvil

sí hubo casos relacionados con autos el problema era el nombre

In [14]:
actividades_autos = [
    "Agencia de autos",
    "Venta de autos usados",
    "Taller de reparación de autos",
    "Repuestos para autos",
    "Alquiler de autos"
]

df_sector_auto = df[
    df["Actividad"].isin(actividades_autos)
].copy()

df_sector_auto.shape

(1259, 17)

In [15]:
reemplazos_texto = {
    "Bater¡a de auto": "Batería de auto",
    "Incumplimiento de garant¡a": "Incumplimiento de garantía",
    "Falta de informaci¢n": "Falta de información",
    "Devoluci¢n de dinero": "Devolución de dinero",
    "Venta enga¤osa": "Venta engañosa",
    "Da¤os y perjuicios": "Daños y perjuicios",
    "Mercanc¡a defectuosa": "Mercancía defectuosa"
}

for col in ["Producto", "Motivo de Queja"]:
    df_sector_auto[col] = df_sector_auto[col].replace(reemplazos_texto)

In [16]:
columnas_texto = [
    "Actividad",
    "Marca de Producto",
    "Producto",
    "Motivo de Queja",
    "Resultado de Queja"
]

for col in columnas_texto:
    print(f"\n--- {col} ---")

    valores_sospechosos = df_sector_auto.loc[
        df_sector_auto[col].astype(str).str.contains(
            r"[¡¢£¤]",
            regex=True,
            na=False
        ),
        col
    ].drop_duplicates()

    print(valores_sospechosos.tolist())


--- Actividad ---
[]

--- Marca de Producto ---
[]

--- Producto ---
[]

--- Motivo de Queja ---
[]

--- Resultado de Queja ---
[]


In [17]:
len(df_sector_auto)

1259

In [18]:
df_sector_auto["Actividad"].value_counts()

,count
Actividad,
Agencia de autos,536
Venta de autos usados,392
Repuestos para autos,197
Taller de reparación de autos,98
Alquiler de autos,36


¿Por qué usamos **Actividad**?
Porque permite identificar y agrupar los registros relacionados con el sector automotriz. Más adelante podrá utilizarse como una variable explicativa para analizar diferencias en el riesgo entre agencias, ventas, talleres, repuestos y alquileres.



# 4. Revisión de valores faltantes del sector automotriz para el dataset df_sector_auto

## 4.1 Cuáles columnas tienen NaN dentro de df_sector_auto

In [19]:
nulos_sector = pd.DataFrame({
    "Cantidad": df_sector_auto.isna().sum(),
    "Porcentaje": (df_sector_auto.isna().mean() * 100).round(2)
})

nulos_sector = nulos_sector[nulos_sector["Cantidad"] > 0]

nulos_sector.sort_values("Cantidad", ascending=False)

,Cantidad,Porcentaje
Efectivo Devuelto al Consumidor (B/.),1238,98.33
Fecha de Cierre de Queja,1014,80.54
Mes Cierre de Queja,1014,80.54
Año Cierre de Queja,1014,80.54
Marca de Producto,53,4.21
Producto,16,1.27




*  Mes Cierre de Queja
*  Fecha de Cierre de Queja
*  Año Cierre de Queja
### tienen los mismo valores NaN lo que puede ser que esos casos esten en PROCESO(todavia no tienen fecha de cierre)


*   Efectivo Devuelto al Consumidor (B/.)
### tiene 1238 NaN. No debemos asumir que todos significan 0; primero hay que comprobar si depende del **Resultado de Queja**.


*   Marca de Producto y Producto
### tienen pocos faltantes, 53 y 16 respectivamente, así que esos sí se puede decidir luego si imputar/eliminar.







## 4.2 Analizar si los valores faltantes tienen una explicación

### 4.2.1 para las fechas:


In [20]:
df_sector_auto.loc[
    df_sector_auto["Fecha de Cierre de Queja"].isna(),
    "Resultado de Queja"
].value_counts()

,count
Resultado de Queja,
En proceso,1014


se confirma que son quejas en proceso...

### 4.2.2 para Efectivo Devuelto al  Consumidor (B/.)

---



In [21]:
df_sector_auto.loc[
    df_sector_auto["Efectivo Devuelto al Consumidor (B/.)"].isna(),
    "Resultado de Queja"
].value_counts() #.sum()

,count
Resultado de Queja,
En proceso,1014
Desistimiento,134
Pendiente constancia desistimiento,27
Acuerdo en audiencia/Varios,18
Ministerio Público,11
Transacción extrajudicial,6
Acuerdo en audiencia/Devolución de dinero,6
Inadmitida,5
Desistimiento/Devolución,3


se confirma que tiene relacion pero nan no es 0, significa que esta columna tiene faltantes reales o casos donde no aplica

revisar casos donde si hay devoluciones de dinero:

In [22]:
df_sector_auto.loc[
    df_sector_auto["Efectivo Devuelto al Consumidor (B/.)"].notna(),
    ["Resultado de Queja", "Efectivo Devuelto al Consumidor (B/.)"]
]

,Resultado de Queja,Efectivo Devuelto al Consumidor (B/.)
206,Acuerdo en audiencia/Devolución de dinero,100.0
368,Acuerdo en audiencia/Devolución de dinero,51.36
369,Acuerdo en audiencia/Devolución de dinero,500.0
575,Desistimiento/Devolución,42.37
1093,Acuerdo en audiencia/Devolución de dinero,79.18
1257,Desistimiento,90.0
2216,Acuerdo en audiencia/Devolución de dinero,100.0
2378,Acuerdo en audiencia/Devolución de dinero,51.36
2379,Acuerdo en audiencia/Devolución de dinero,500.0
2585,Desistimiento/Devolución,42.37


In [23]:
df_sector_auto["Efectivo Devuelto al Consumidor (B/.)"].notna().sum()

np.int64(21)

esta columna solo tiene 21 registros de 1269...

In [24]:
total = len(df_sector_auto)

con_dato = df_sector_auto["Efectivo Devuelto al Consumidor (B/.)"].notna().sum()
sin_dato = df_sector_auto["Efectivo Devuelto al Consumidor (B/.)"].isna().sum()

porcentaje_con_dato = (con_dato / total) * 100
porcentaje_sin_dato = (sin_dato / total) * 100

print("Total de registros:", total)
print("Con dato:", con_dato, f"({porcentaje_con_dato:.2f}%)")
print("Sin dato:", sin_dato, f"({porcentaje_sin_dato:.2f}%)")

Total de registros: 1259
Con dato: 21 (1.67%)
Sin dato: 1238 (98.33%)


con un  porcentaje de 98% se pensaria que es mejor descartarla pero es mejor prevenir y hacer una analisis

### 4.2.3 para Marca de producto y Producto

In [25]:
df_sector_auto.loc[
    df_sector_auto["Marca de Producto"].isna(),
    ["Actividad", "Producto", "Motivo de Queja", "Resultado de Queja"]
].head(20)

,Actividad,Producto,Motivo de Queja,Resultado de Queja
4131,Agencia de autos,Auto,Incumplimiento de garantía,En proceso
4144,Repuestos para autos,NaN,Devolución de dinero,Desistimiento
4147,Repuestos para autos,NaN,Devolución de dinero,En proceso
4375,Agencia de autos,NaN,Veracidad de la publicidad,Desistimiento
4395,Alquiler de autos,Auto,Falta de información,Desistimiento
4494,Venta de autos usados,Auto,Incumplimiento de garantía,En proceso
4532,Repuestos para autos,NaN,Incumplimiento de servicio,En proceso
4568,Repuestos para autos,Batería de auto,Incumplimiento de garantía,En proceso
4686,Agencia de autos,Auto,Incumplimiento de garantía,En proceso
4687,Agencia de autos,Auto,Incumplimiento de garantía,En proceso


In [26]:
df_sector_auto.loc[
    df_sector_auto["Producto"].isna(),
    ["Actividad", "Marca de Producto", "Motivo de Queja", "Resultado de Queja"]
].head(20)

,Actividad,Marca de Producto,Motivo de Queja,Resultado de Queja
4144,Repuestos para autos,NaN,Devolución de dinero,Desistimiento
4147,Repuestos para autos,NaN,Devolución de dinero,En proceso
4375,Agencia de autos,NaN,Veracidad de la publicidad,Desistimiento
4532,Repuestos para autos,NaN,Incumplimiento de servicio,En proceso
4836,Repuestos para autos,NaN,Incumplimiento de servicio,En proceso
4981,Repuestos para autos,NaN,Daños y perjuicios,Pendiente constancia desistimiento
4991,Repuestos para autos,NaN,Incumplimiento de garantía,En proceso
5105,Agencia de autos,NaN,Devolución de dinero,En proceso
5148,Repuestos para autos,NaN,Incumplimiento de garantía,En proceso
5314,Repuestos para autos,NaN,Custodia de bien,En proceso


En varios casos, aunque **Producto/Marca de Producto** esté vacío, la **Actividad** ya indica claramente que el registro pertenece al sector automotriz, por ejemplo Repuestos para autos o Agencia de autos

se decidio reemplazar esos NaN por algo mas descriptivo:

In [27]:
df_sector_auto["Marca de Producto"] = df_sector_auto["Marca de Producto"].fillna("No registrado")

df_sector_auto["Producto"] = df_sector_auto["Producto"].fillna("No registrado")

In [28]:
df_sector_auto[["Marca de Producto", "Producto"]].isna().sum()

,0
Marca de Producto,0
Producto,0


In [29]:
df_sector_auto[
    ["Actividad", "Marca de Producto", "Producto", "Motivo de Queja", "Resultado de Queja"]
].head(50)

,Actividad,Marca de Producto,Producto,Motivo de Queja,Resultado de Queja
184,Agencia de autos,Chevrolet,Auto,Incumplimiento de garantía,En proceso
188,Venta de autos usados,No se Registro,Auto,Falta de información,Desistimiento
189,Agencia de autos,Toyota,Auto,Incumplimiento de garantía,Acuerdo en audiencia/Varios
191,Venta de autos usados,No se Registro,No se Registro,Incumplimiento de garantía,En proceso
192,Repuestos para autos,No se Registro,Batería de auto,Incumplimiento de garantía,En proceso
193,Repuestos para autos,No se Registro,Pieza de Repuesto,Falta de información,En proceso
194,Repuestos para autos,Hyundai,Auto,Incumplimiento de garantía,En proceso
195,Agencia de autos,Toyota,Auto,Incumplimiento de garantía,En proceso
196,Venta de autos usados,Hyundai,Auto,Falta de información,Ministerio Público
197,Agencia de autos,No se Registro,Auto,Incumplimiento de garantía,Inadmitida


## 5. Qué columnas vamos a conservar para construir la clasificación de riesgo Alto / Medio / Bajo y cuáles no nos aportan al modelo?**texto en negrita**

### 5.3.1 Separar las columnas utiles

In [30]:
columnas_utiles = [
    "Actividad",
    "Marca de Producto",
    "Producto",
    "Motivo de Queja",
    "Velor de Apertura de Queja",
    "Resultado de Queja"
]

In [31]:
df_modelo = df_sector_auto[columnas_utiles].copy()

In [32]:
df_modelo.head()

,Actividad,Marca de Producto,Producto,Motivo de Queja,Velor de Apertura de Queja,Resultado de Queja
184,Agencia de autos,Chevrolet,Auto,Incumplimiento de garantía,23648.0,En proceso
188,Venta de autos usados,No se Registro,Auto,Falta de información,3200.0,Desistimiento
189,Agencia de autos,Toyota,Auto,Incumplimiento de garantía,12500.0,Acuerdo en audiencia/Varios
191,Venta de autos usados,No se Registro,No se Registro,Incumplimiento de garantía,4500.0,En proceso
192,Repuestos para autos,No se Registro,Batería de auto,Incumplimiento de garantía,90.9,En proceso


In [33]:
df_modelo.shape

(1259, 6)

# 6. Cómo se distribuyen estas columnas?, sobre todo Motivo de Queja, Valor de Apertura de Queja y Resultado de Queja, para decidir con qué lógica construiremos el riesgo.

In [34]:
df_modelo["Motivo de Queja"].value_counts()

,count
Motivo de Queja,
Incumplimiento de garantía,841
Falta de información,197
Incumplimiento de servicio,87
Devolución de dinero,35
Custodia de bien,34
Vicio oculto,22
Incumplimiento de contrato,20
Cobro indebido,9
Daños y perjuicios,5


In [35]:
df_modelo["Resultado de Queja"].value_counts()

,count
Resultado de Queja,
En proceso,1014
Desistimiento,137
Pendiente constancia desistimiento,27
Acuerdo en audiencia/Devolución de dinero,20
Acuerdo en audiencia/Varios,18
Ministerio Público,11
Desistimiento/Devolución,7
Transacción extrajudicial,6
Inadmitida,5


In [36]:
df_modelo["Velor de Apertura de Queja"].describe()

,Velor de Apertura de Queja
count,1259.0
unique,840.0
top,500.0
freq,11.0


In [37]:
df_modelo["Velor de Apertura de Queja"].dtype

dtype('O')

segun el describe y el .dtype parece estar tratándolo como texto y no como número.

## 6.1 Convertir el valor de apertura a un tipo numerico

In [38]:
df_modelo["Velor de Apertura de Queja"] = pd.to_numeric(
    df_modelo["Velor de Apertura de Queja"],
    errors="coerce"
)

In [39]:
df_modelo["Velor de Apertura de Queja"].dtype

dtype('float64')

In [40]:
df_modelo["Velor de Apertura de Queja"].describe()

,Velor de Apertura de Queja
count,899.000000
mean,8164.561090
std,9100.145694
min,0.000000
25%,351.500000
50%,3950.000000
75%,15900.000000
max,30000.000000


In [41]:
print("Total de registros:", len(df_modelo))

print(
    "Valores numéricos válidos:",
    df_modelo["Velor de Apertura de Queja"].notna().sum()
)

print(
    "Valores NaN:",
    df_modelo["Velor de Apertura de Queja"].isna().sum()
)

Total de registros: 1259
Valores numéricos válidos: 899
Valores NaN: 360


In [42]:
conversion = pd.to_numeric(
    df_sector_auto["Velor de Apertura de Queja"],
    errors="coerce"
)

valores_no_convertidos = df_sector_auto.loc[
    conversion.isna() &
    df_sector_auto["Velor de Apertura de Queja"].notna(),
    "Velor de Apertura de Queja"
]

valores_no_convertidos.value_counts()

,count
Velor de Apertura de Queja,
"7,000.00",5
"25,000.00",4
"5,500.00",4
"1,000.00",4
"8,000.00",4
...,...
"20,204.34",1
"25,490.00",1
"26,739.30",1


lo esta tomando como NaN porque tiene ,

In [43]:
df_modelo["Velor de Apertura de Queja"] = (
    df_sector_auto["Velor de Apertura de Queja"]
    .astype(str)
    .str.replace(",", "", regex=False)
)

In [44]:
df_modelo["Velor de Apertura de Queja"] = pd.to_numeric(
    df_modelo["Velor de Apertura de Queja"],
    errors="coerce"
)

In [45]:
print("Válidos:", df_modelo["Velor de Apertura de Queja"].notna().sum())
print("NaN:", df_modelo["Velor de Apertura de Queja"].isna().sum())

Válidos: 1259
NaN: 0


# 7. Observar los cuartiles del monto(asi no se inventan los límites)

In [46]:
df_modelo["Velor de Apertura de Queja"].quantile([0.25, 0.50, 0.75])

,Velor de Apertura de Queja
0.25,777.915
0.50,6200.000
0.75,18123.000


# 8. Variable objetivo: Nivel de Riesgo (Bajo / Medio / Alto) Diseñar reglas de puntuacion

## 8.1 clasificar los motivos de queja por nivel de impacto

In [55]:
riesgo_motivo = {
    "Falta de información": 0,

    "Venta engañosa": 1,
    "Incumplimiento de servicio": 1,
    "Incumplimiento de contrato": 1,
    "Custodia de bien": 1,
    "Cobro indebido": 1,
    "Veracidad de la publicidad": 1,

    "Incumplimiento de garantía": 2,
    "Devolución de dinero": 2,
    "Daños y perjuicios": 2,
    "Mercancía defectuosa": 2,
    "Vicio oculto": 2
}



*  0 = impacto bajo
*  1 = impacto medio
*  2 = impacto alto




## 8.2 Asignar puntuación según Resultado de Queja

In [56]:
riesgo_resultado = {
    # 0 puntos - favorable
    "Acuerdo en audiencia/Devolución de dinero": 0,
    "Acuerdo en audiencia/Reparación": 0,
    "Acuerdo en audiencia/Aclaración": 0,
    "Acuerdo en audiencia/Varios": 0,
    "Ordena corrección": 0,

    # 1 punto - pendiente o sin resolución definitiva
    "En proceso": 1,
    "Pendiente constancia desistimiento": 1,
    "Desistimiento": 1,
    "Desistimiento/Devolución": 1,
    "Transacción extrajudicial": 1,
    "Abandono": 1,

    # 2 puntos - escalado o desfavorable
    "Ministerio Público": 2,
    "Corregiduría/Casa de Justicia Comunitaria de Paz": 2,
    "Inadmitida": 2,
    "Inhibe": 2,
    "Caducidad de la instancia": 2,
    "Resolución no acceder": 2
}

In [57]:
df_modelo["Puntos_Resultado"] = df_modelo["Resultado de Queja"].map(riesgo_resultado)

¿Hay algún Resultado de Queja al que olvidamos asignarle puntos?:

In [58]:
df_modelo.loc[
    df_modelo["Puntos_Resultado"].isna(),
    "Resultado de Queja"
].value_counts()

,count
Resultado de Queja,


In [59]:
df_modelo[
    ["Resultado de Queja", "Puntos_Resultado"]
].head(20)

,Resultado de Queja,Puntos_Resultado
184,En proceso,1
188,Desistimiento,1
189,Acuerdo en audiencia/Varios,0
191,En proceso,1
192,En proceso,1
193,En proceso,1
194,En proceso,1
195,En proceso,1
196,Ministerio Público,2
197,Inadmitida,2


## 8.3: asignar puntos según el monto reclamado

In [60]:
def puntuar_monto(valor):
    if valor <= 780.20:
        return 0
    elif valor <= 18298:
        return 1
    else:
        return 2

df_modelo["Puntos_Monto"] = df_modelo["Velor de Apertura de Queja"].apply(puntuar_monto)

In [61]:
df_modelo[
    ["Velor de Apertura de Queja", "Puntos_Monto"]
].head(20)

,Velor de Apertura de Queja,Puntos_Monto
184,23648.00,2
188,3200.00,1
189,12500.00,1
191,4500.00,1
192,90.90,0
193,250.00,0
194,260.00,0
195,15900.00,1
196,8800.00,1
197,367.08,0


In [62]:
df_modelo["Puntos_Motivo"] = df_modelo["Motivo de Queja"].map(riesgo_motivo)

In [63]:
df_modelo[
    ["Motivo de Queja", "Puntos_Motivo"]
].head(20)

,Motivo de Queja,Puntos_Motivo
184,Incumplimiento de garantía,2
188,Falta de información,0
189,Incumplimiento de garantía,2
191,Incumplimiento de garantía,2
192,Incumplimiento de garantía,2
193,Falta de información,0
194,Incumplimiento de garantía,2
195,Incumplimiento de garantía,2
196,Falta de información,0
197,Incumplimiento de garantía,2


In [64]:
df_modelo["Puntaje_Total"] = (
    df_modelo["Puntos_Motivo"] +
    df_modelo["Puntos_Resultado"] +
    df_modelo["Puntos_Monto"]
)

In [65]:
df_modelo[
    [
        "Motivo de Queja",
        "Puntos_Motivo",
        "Resultado de Queja",
        "Puntos_Resultado",
        "Velor de Apertura de Queja",
        "Puntos_Monto",
        "Puntaje_Total"
    ]
].head(20)

,Motivo de Queja,Puntos_Motivo,Resultado de Queja,Puntos_Resultado,Velor de Apertura de Queja,Puntos_Monto,Puntaje_Total
184,Incumplimiento de garantía,2,En proceso,1,23648.00,2,5
188,Falta de información,0,Desistimiento,1,3200.00,1,2
189,Incumplimiento de garantía,2,Acuerdo en audiencia/Varios,0,12500.00,1,3
191,Incumplimiento de garantía,2,En proceso,1,4500.00,1,4
192,Incumplimiento de garantía,2,En proceso,1,90.90,0,3
193,Falta de información,0,En proceso,1,250.00,0,1
194,Incumplimiento de garantía,2,En proceso,1,260.00,0,3
195,Incumplimiento de garantía,2,En proceso,1,15900.00,1,4
196,Falta de información,0,Ministerio Público,2,8800.00,1,3
197,Incumplimiento de garantía,2,Inadmitida,2,367.08,0,4


In [66]:
def clasificar_riesgo(puntaje):
    if puntaje <= 2:
        return "Bajo"
    elif puntaje <= 4:
        return "Medio"
    else:
        return "Alto"

df_modelo["Nivel_Riesgo"] = df_modelo["Puntaje_Total"].apply(clasificar_riesgo)

In [67]:
df_modelo["Nivel_Riesgo"].value_counts()

,count
Nivel_Riesgo,
Medio,709
Alto,292
Bajo,258


In [68]:
df_modelo.groupby("Nivel_Riesgo")["Velor de Apertura de Queja"].describe()

,count,mean,std,min,25%,50%,75%,max
Nivel_Riesgo,,,,,,,,
Alto,292.0,22787.637842,4660.410344,1175.13,20488.00,22900.00,25923.75,30000.0
Bajo,258.0,2084.643953,3259.442674,0.00,180.00,500.00,3000.00,18180.0
Medio,709.0,6760.058307,6500.574512,0.00,950.06,4627.86,11995.00,29050.0


In [69]:
pd.crosstab(
    df_modelo["Nivel_Riesgo"],
    df_modelo["Motivo de Queja"]
)

Motivo de Queja,Cobro indebido,Custodia de bien,Daños y perjuicios,Devolución de dinero,Falta de información,Incumplimiento de contrato,Incumplimiento de garantía,Incumplimiento de servicio,Mercancía defectuosa,Venta engañosa,Veracidad de la publicidad,Vicio oculto
Nivel_Riesgo,,,,,,,,,,,,
Alto,0,0,0,0,0,0,286,0,0,0,0,6
Bajo,6,19,0,0,179,7,10,35,0,1,1,0
Medio,3,15,5,35,18,13,545,52,2,3,2,16


In [70]:
df_modelo["Marca de Producto"].drop_duplicates().sort_values().tolist()

['#\xadNULO!',
 'Acura',
 'Audi',
 'Ava',
 'BMW',
 'BMX',
 'Bestune',
 'Changan',
 'Cherry',
 'Chery QQ',
 'Chevrolet',
 'DFSK',
 'Daihatsu',
 'Dodge',
 'Dongfeng',
 'Fiat',
 'Ford',
 'Forthing',
 'GAC',
 'GAP',
 'Geely',
 'Haval',
 'Honda',
 'Hyundai',
 'Infiniti',
 'Isuzu',
 'JAC',
 'JEEP',
 'Jetour',
 'KIA',
 'KTM',
 'Kaiyi',
 'Kamura',
 'Karry',
 'Keeway',
 'Land Rover',
 'Lexus',
 'MG',
 'Maxus',
 'Mazda',
 'Mercedez Benz',
 'Mini Cooper',
 'Mitsubishi',
 'Nissan',
 'No registrado',
 'No se Registro',
 'Omoda',
 'POER',
 'Panasonic',
 'Peugeot',
 'Porsche',
 'RAM',
 'Renault',
 'Subaru',
 'Suzuki',
 'Tasco',
 'Tesla',
 'Toyota',
 'VGV',
 'Volkswagen',
 'Volvo',
 'Yamaha',
 'Yema']

In [71]:
df_modelo.loc[
    df_modelo["Marca de Producto"] == "Panasonic",
    ["Actividad", "Marca de Producto", "Producto", "Motivo de Queja"]
]

,Actividad,Marca de Producto,Producto,Motivo de Queja
7109,Agencia de autos,Panasonic,Batería de auto,Incumplimiento de garantía


In [72]:
df_modelo[
    ["Actividad", "Marca de Producto", "Producto"]
].sort_values(["Marca de Producto", "Actividad"])

,Actividad,Marca de Producto,Producto
5174,Repuestos para autos,#­NULO!,#­NULO!
7137,Venta de autos usados,Acura,Automóvil
6606,Repuestos para autos,Audi,Automóvil
6769,Taller de reparación de autos,Audi,Automóvil
6902,Taller de reparación de autos,Audi,Automóvil
...,...,...,...
6609,Taller de reparación de autos,Volvo,Automóvil
6613,Taller de reparación de autos,Volvo,Automóvil
6623,Venta de autos usados,Volvo,Amortiguadores
6932,Taller de reparación de autos,Yamaha,Moto


In [73]:
df_marcas_autos = df_modelo[
    df_modelo["Producto"].isin(["Auto", "Automóvil"])
].copy()

In [74]:
df_modelo[
    [
        "Marca de Producto",
        "Producto",
        "Motivo de Queja",
        "Puntos_Motivo",
        "Resultado de Queja",
        "Puntos_Resultado",
        "Velor de Apertura de Queja",
        "Puntos_Monto",
        "Puntaje_Total",
        "Nivel_Riesgo"
    ]
]

,Marca de Producto,Producto,Motivo de Queja,Puntos_Motivo,Resultado de Queja,Puntos_Resultado,Velor de Apertura de Queja,Puntos_Monto,Puntaje_Total,Nivel_Riesgo
184,Chevrolet,Auto,Incumplimiento de garantía,2,En proceso,1,23648.0,2,5,Alto
188,No se Registro,Auto,Falta de información,0,Desistimiento,1,3200.0,1,2,Bajo
189,Toyota,Auto,Incumplimiento de garantía,2,Acuerdo en audiencia/Varios,0,12500.0,1,3,Medio
191,No se Registro,No se Registro,Incumplimiento de garantía,2,En proceso,1,4500.0,1,4,Medio
192,No se Registro,Batería de auto,Incumplimiento de garantía,2,En proceso,1,90.9,0,3,Medio
...,...,...,...,...,...,...,...,...,...,...
7850,Kaiyi,Automóvil,Incumplimiento de garantía,2,En proceso,1,26739.3,2,5,Alto
7851,Hyundai,Automóvil,Incumplimiento de garantía,2,En proceso,1,5100.0,1,4,Medio
7954,GAC,Automóvil,Incumplimiento de servicio,1,En proceso,1,24950.0,2,4,Medio
7955,No se Registro,Pieza de Repuesto,Custodia de bien,1,En proceso,1,353.0,0,2,Bajo


In [75]:
pd.set_option("display.max_rows", None)

In [76]:
df_sin_marca = df_modelo[
    ~df_modelo["Marca de Producto"].isin(["No se Registro", "No registrado"])
]

len(df_sin_marca)

1065

In [77]:
print("Total actual:", len(df_modelo))
print("Sin marca registrada:", len(df_modelo) - len(df_sin_marca))
print("Registros restantes:", len(df_sin_marca))

Total actual: 1259
Sin marca registrada: 194
Registros restantes: 1065
